splunk-otel-android/src/main/java/com/splunk/rum/NamedTrackableTracer.java

```diff
@@ -119,13 +135,14 @@ public void endSpanForActivityResumed() {
 
     @Override
     public void endActiveSpan() {
+        SplunkRum.getStartupTimer().end();
```

Comment:
this is a little bit of a bummer. I'd like to figure out a way to know that we're in the right kind of span when we do this, rather than blindly ending it always.

Reply:
I added a null check on the startup span, but it's still a bit ugly to blindly ending the startup span here, when there should only ever be one.

Reply:
> I'd like to figure out a way to know that we're in the right kind of span when we do this, rather than blindly ending it always.

I wonder if the current problem might be caused by the fact that `NamedTrackableTracer`'s supposed to be completely independent of the lifecycle phases, and yet at the same time it must keep track of whether the current execution is an application start, whether it's a cold or hot one. Just thinking out loud, but what if we had separate instances (and implementations) for each lifecycle? When you `endActiveSpan()` the next time you create one a different implementation of the tracer might get chosen, depending on what lifecycle callback gets called.

Reply:
That's a possibility. It would lead to more allocations, rather than just one per Activity/Fragment, but it's definitely worth a try. I don't want to try it in this PR, but I like the suggestion.